src/wsdl/elements.ts

```diff
@@ -229,16 +229,16 @@ export class ElementElement extends Element {
 
     const isMany = maxOccurs > 1;
 
-    if (isMany) {
+    if (isMany && name) {
       name += '[]';
     }
 
     if (xmlns && xmlns[TNS_PREFIX]) {
       this.$targetNamespace = xmlns[TNS_PREFIX];
     }
-    let type: any = this.$type || this.$ref;
-    if (type) {
-      type = splitQName(type);
+    const inferredType: string = this.$type || this.$ref;
```

Comment:
What was the reason to rename this variable? Usually this is not a good practice to rename variables/function names without actual changes. 

Also, not sure that that type `string` is correct here? Is it possible that `$type` and `$ref` can be undefined or null?

I think this change is not related and should be reverted.

Reply:
> What was the reason to rename this variable? Usually this is not a good practice to rename variables/function names without actual changes.

I did this to better debug and can remove it. Main point is that type is assigned a string value and then is assigned to an object.

Reply:
thanks